src/lib/transaction_snark/transaction_snark.mli

```diff
@@ -31,45 +31,222 @@ module Pending_coinbase_stack_state : sig
     [@@deriving sexp, hash, compare, yojson]
   end
 
+  module Poly : sig
+    [%%versioned:
+    module Stable : sig
+      module V1 : sig
+        type 'pending_coinbase t =
+          {source: 'pending_coinbase; target: 'pending_coinbase}
+        [@@deriving compare, eq, fields, hash, sexp, yojson]
+
+        val to_latest :
```

Comment:
should we be doing this sort of thing in general, for versioned types with parameters?

Reply:
It's good practice for when we change things, but I don't think we strictly need to. It's easy enough to build them out as we go.